pkg/apis/ceph.rook.io/v1/network.go

```diff
@@ -16,13 +16,70 @@ limitations under the License.
 
 package v1
 
+import (
+	"fmt"
+	"net"
+
+	"github.com/pkg/errors"
+)
+
 // IsMultus get whether to use multus network provider
 func (n *NetworkSpec) IsMultus() bool {
-	return n.Provider == "multus"
+	return n.Provider == NetworkProviderMultus
 }
 
 // IsHost get whether to use host network provider. This method also preserve
 // compatibility with the old HostNetwork field.
 func (n *NetworkSpec) IsHost() bool {
-	return (n.HostNetwork && n.Provider == "") || n.Provider == "host"
+	return (n.HostNetwork && n.Provider == NetworkProviderDefault) || n.Provider == NetworkProviderHost
+}
+
+func ValidateNetworkSpec(spec NetworkSpec) error {
+	if spec.IsMultus() && len(spec.Selectors) == 0 {
+		return errors.Errorf("at least one network selector must be specified when using the %q network provider", NetworkProviderMultus)
+	}
+
+	if !spec.NetworkRanges.IsEmpty() && !spec.IsMultus() && !spec.IsHost() {
+		// TODO: be sure to update docs that networkRanges can be specified for host networking  as
+		// well as multus so that the override configmap doesn't need to be set
+		return errors.Errorf("network ranges can only be specified for %q and %q network providers", NetworkProviderHost, NetworkProviderMultus)
+	}
+
+	if err := spec.NetworkRanges.Validate(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func ValidateNetworkSpecUpdate(oldSpec, newSpec NetworkSpec) error {
+	// Allow an attempt to enable or disable host networking, but not other provider changes
+	oldProvider := oldSpec.Provider
+	newProvider := newSpec.Provider
+	if oldProvider != newProvider && oldProvider != "host" && newProvider != "host" {
```

Comment:
```suggestion
	if oldProvider != newProvider && oldProvider != "host" && newProvider != "host" {
```

I'm confused understanding this condition, could you explain here in discussion?

Reply:
This is copy-pasted from some of @travisn 's code. Maybe he has a better explanation

Reply:
Hopefully the comment on line 79 explains it:
```
// Allow an attempt to enable or disable host networking, but not other provider changes
```